atlp-rwanda/ecommerce-app-champions-bn
# ft(social-account):login via google

##  what does this PR do:
sign in with social account
##  Descriptions
- Implement Google OAuth2 authentication for users to log in
- Authenticate users with Google and create a new account
- Redirect users to /welcome endpoint upon successful authentication
- Allow users to log in again using their Google account
- Generate and document Swagger documentation for frontend team
## How this should be tested
- clone the repo
- type git checkout ft-social-account
- type npm install to install dependencies
- type  npm run migrate:up to run migration
-  type npm run dev to start server
- To test login via google by using browser 
copy and paste `GOOGLE_CLIENT_ID=''
GOOGLE_SECRET=''`. into your  .env file
Note: to test this make sure your app is running on port 
open your browser   then type `http://localhost:port/auth/google`
once you done  click on sign in look into your database
- To test login via google by using swagger
  open your browser   then type `http://localhost:port/docs`
click on Authorize
then copy and paste `GOOGLE_CLIENT_ID=''
GOOGLE_SECRET=''`
Note:remember to check the 
 checkbox and then click Authorize
## Sign in with social account card #9
